contracts/andromeda_timelock/src/contract.rs

```diff
@@ -143,36 +158,31 @@ fn execute_release_funds(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
+    recipient_addr: Option<String>,
+    start_after: Option<String>,
+    limit: Option<u32>,
 ) -> Result<Response, ContractError> {
-    let result: Option<Escrow> = get_funds(deps.storage, info.sender.to_string())?;
+    let recipient_addr = recipient_addr.unwrap_or_else(|| info.sender.to_string());
```

Comment:
Another approach for withdrawing is to require the caller to specify an owner and recipient they wish to withdraw for, rather than here where we try to withdraw all of them. This would be more performant if we expect the caller to always know the exact escrow they want. Thoughts?